### PR TITLE
Scott evgeni/311 replace authorityclient with networkclient

### PR DESCRIPTION
**Pull request by BatiGencho**
This is the scope of the ticket:

https://github.com/botanix-labs/botanix/issues/311

### CODE REVIEW

File: crates/consensus/authority/src/lib.rs [hunk 1]
```diff
@@ -151,45 +158,36 @@ impl Storage {
...
-    pub(crate) headers: HashMap<BlockNumber, Header>,
+/// In-memory storage for the chain the authority seal engine is building.
+pub(crate) struct StorageInner<Client> {
+    client: Client,
     /// A mapping between block hash and number.
-    pub(crate) hash_to_number: HashMap<BlockHash, BlockNumber>,
-    /// Bodies buffered for download.
-    pub(crate) bodies: HashMap<BlockHash, BlockBody>,
     /// Tracks best block
     pub(crate) best_block: u64,
```
Comment (arminsabouri) [inline crates/consensus/authority/src/lib.rs]: Is it worth keeping best_block and best_hash? Can we get these from the Client?

File: crates/consensus/authority/src/lib.rs [hunk 2]
```diff
@@ -209,26 +207,26 @@ pub(crate) struct StorageInner {
...
-            BlockHashOrNumber::Number(num) => num,
-        };
-        self.headers.get(&num).cloned()
+        self.client.header_by_hash_or_number(hash_or_num).unwrap_or(None)
     }
 
-    /// Inserts a new header+body pair
-    pub(crate) fn insert_new_block(&mut self, mut header: Header, body: BlockBody) {
+    /// Inserts a new header pair
+    pub(crate) fn insert_new_block(&mut self, mut header: Header) {
```
Comment (arminsabouri) [inline crates/consensus/authority/src/lib.rs]: Can we remove `insert_new_block`?

**Review by arminsabouri — COMMENTED** [review]
On the right track. Just had a couple questions

File: crates/consensus/authority/src/lib.rs (hunk 1 above)
Comment (BatiGencho) [reply crates/consensus/authority/src/lib.rs]: Yep, just pushed a commit that does that. Please check it

File: crates/consensus/authority/src/lib.rs [hunk 3]
```diff
@@ -186,11 +185,6 @@ where
 /// In-memory storage for the chain the authority seal engine is building.
 pub(crate) struct StorageInner<Client> {
     client: Client,
-    /// A mapping between block hash and number.
-    /// Tracks best block
-    pub(crate) best_block: u64,
-    /// Tracks hash of best block
-    pub(crate) best_hash: B256,
     /// The total difficulty of the chain until this block
     pub(crate) total_difficulty: U256,
```
Comment (arminsabouri) [inline crates/consensus/authority/src/lib.rs]: `total_difficulty` could get removed in a future PR

File: crates/consensus/authority/src/lib.rs (hunk 2 above)
Comment (BatiGencho) [reply crates/consensus/authority/src/lib.rs]: done

File: crates/consensus/authority/src/lib.rs [hunk 4]
```diff
@@ -224,16 +218,31 @@ where
...
+    pub(crate) fn get_best_block_and_hash(
+        &self,
+    ) -> Result<(u64, FixedBytes<32>), BlockExecutionError> {
+        let best_block =
+            self.client.best_block_number().map_err(|_| BlockExecutionError::ProviderError)?;
+        let best_hash = self
+            .client
+            .block_hash(best_block)
+            .map_err(|_| BlockExecutionError::ProviderError)?
+            .ok_or(BlockExecutionError::MissingBlockHash)?;
```
Comment (arminsabouri) [inline crates/consensus/authority/src/lib.rs]: Should panic! here

**Review by arminsabouri — APPROVED** [review]

File: crates/consensus/authority/src/lib.rs (hunk 4 above)
Comment (BatiGencho) [reply crates/consensus/authority/src/lib.rs]: added

**Review by arminsabouri — APPROVED** [review]

**Pull request merged**